taggify.js

```diff
@@ -0,0 +1,696 @@
+          }
+          if (widths[i-1] < mouseX && mouseX < widths[i]) {
+            fillInput(input, start_tags_container);
+          }
+          console.log("Click was on the ~"+counter+" tag");
+          break;
+        }
+      }
+      if (widths[widths.length-2] < mouseX && mouseX < widths[widths.length-1]) {
+        console.log("Click was on the last tag")
+        if (container == end_tags_container) {
+          moveElementsTo(start_tags_container, null)
+        }
+        fillInput(input, start_tags_container);
+      }
+    } else if (container == start_tags_container && mouseX < widths[0]) {
+      console.log("Click was before the start of the start tag");
+      moveElementsTo(end_tags_container, null);
+    } else if (container == end_tags_container && mouseX > widths[widths.length-1]) {
+      console.log("Click was afte the end of the end tags");
+      if (SETTINGS.automatically_start_editing_last_tag) {
+        moveElementsTo(start_tags_container, null);
+      }
+    }
+    input.focus();
+  }
+
+  function taggify(tag_input) {
+    var start_tags_container = tag_input.children[0];
+    var input = tag_input.children[1];
+    var end_tags_container = tag_input.children[2];
+    var autocomplete = tag_input.children[3];
+
+    // Delete tag handlers
+    // These need to go first to override the tag onclick handlers
+
+    start_tags_container.addEventListener('click', function(e) {
+      var start_tags_container = this.parentElement.children[0];
+      var input = this.parentElement.children[1];
+      var end_tags_container = this.parentElement.children[2];
+
+      var start_tags = start_tags_container.children;
+      var end_tags = end_tags_container.children;
+
+      var tag = e.target;
+
+      if (tag.classList.contains('delete')) {
+        tag = tag.parentElement;
+        tag.parentElement.removeChild(tag);
+        input.focus();
+      }
+    });
+
+    end_tags_container.addEventListener('click', function(e) {
+      var start_tags_container = this.parentElement.children[0];
+      var input = this.parentElement.children[1];
+      var end_tags_container = this.parentElement.children[2];
+
+      var start_tags = start_tags_container.children;
+      var end_tags = end_tags_container.children;
+
+      var tag = e.target;
+
+      if (tag.classList.contains('delete')) {
+        tag = tag.parentElement;
+        tag.parentElement.removeChild(tag);
+        input.focus();
+      }
+    });
+
+    // Container click listeners
+    // These are for when you click on/between/over/around a tag,
+    // getting the tag to actually become editable.
+
+    start_tags_container.addEventListener('click', function(e) {
+      handleContainerClick(this, e.clientX);
+    });
+
+    end_tags_container.addEventListener('click', function(e) {
+      handleContainerClick(this, e.clientX);
+    });
+
+    // This is a fallback for the start_tags_container and end_tags_container click handlers -- some parts
```

Comment:
Good comment :+1:! Hard to tell what the code below is there for from reading it, so it's good to have this explanation.